core/Azure.Mcp.Core/src/Services/Azure/Authentication/HttpOnBehalfOfTokenCredentialProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Azure.Core;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Identity.Web;
+
+namespace Azure.Mcp.Core.Services.Azure.Authentication;
+
+public class HttpOnBehalfOfTokenCredentialProvider : IAzureTokenCredentialProvider
+{
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly ILogger<HttpOnBehalfOfTokenCredentialProvider> _logger;
+
+    public HttpOnBehalfOfTokenCredentialProvider(
+        IHttpContextAccessor httpContextAccessor,
+        ILogger<HttpOnBehalfOfTokenCredentialProvider> logger)
+    {
+        _httpContextAccessor = httpContextAccessor;
+        _logger = logger;
+    }
+
+    /// <inheritdoc/>
+    public Task<TokenCredential> GetTokenAsync(string? tenantId, CancellationToken cancellationToken)
+    {
+        if (_httpContextAccessor.HttpContext is not HttpContext httpContext)
+        {
+            throw new InvalidOperationException("There is no ongoing HTTP request.");
+        }
+
+        if (tenantId is not null)
+        {
+            if (httpContext.User.FindFirst("tid")?.Value is string tidClaim
```

Comment:
Wondering - are we guaranteed to only receive v2.0 tokens here, if not should be consider fall back to the legacy claim URI?

```C#
var principal = httpContext.User;
principal.FindFirst("tid")?.Value
    ?? principal.FindFirst("http://schemas.microsoft.com/identity/claims/tenantid")?.Value;
```